lib/zendesk_apps_support/validations/requirements.rb

```diff
@@ -121,6 +134,15 @@ def invalid_requirements_types(requirements)
                                 count: invalid_types.length)
           end
         end
+
+        def validate_custom_objects_keys(keys, expected_keys, identifier, errors = [])
+          missing_keys = expected_keys - keys
+          missing_keys.each do |key|
+            errors << ValidationError.new(:missing_required_fields,
+                                          field: key,
+                                          identifier: identifier)
+          end
```

Comment:
I don't think `errors` is being returned here.

<img width="784" alt="Screen Shot 2019-11-04 at 11 08 58 am" src="https://user-images.githubusercontent.com/17760485/68094347-9dd4fc80-fef3-11e9-871a-bf614f0b6f59.png">

Reply:
I can suggest another approach, which is to either use a dot inject/each with object do, append, and flatten. Something like this. 

<img width="889" alt="Screen Shot 2019-11-04 at 11 18 19 am" src="https://user-images.githubusercontent.com/17760485/68094476-f22cac00-fef4-11e9-93db-2cf6de6781b8.png">

Reply:
This pattern is a bit confusing but what's happening is the errors object is getting instantiated and eventually returned with the `tap` call, then getting passed down into these methods which is pushing the errors onto it.
I don't love this pattern as it's a bit confusing however this seems to be being done in a few places in the code base.

Reply:
I understand 🤢And I also get what you mean about the existing pattern.

This is a **private method that accepts errors as an argument** with default value `[]`.

Does it mean that we do not need the third argument declared? 

Does it mean that we have `@errors` all along?

Reply:
Not quite, its not the `@erros` that's being passed, its the `errors` var initialised in the `invalid_custom_objects` method during `[].tap | errors |`
Let's 🍖 about this but yeah, probably a pattern we should like into changing sooner rather than later